x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.test.ts

```diff
@@ -0,0 +1,350 @@
+      .index as jest.Mock).mock.calls[0][0].body;
+    expect(actionDoc.data.command).toEqual('unisolate');
+  });
+
+  describe('License Level', () => {
+    it('allows platinum license levels to isolate hosts', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        license: Platinum,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+    it('prohibits license levels less than platinum from isolating hosts', async () => {
+      licenseEmitter.next(Gold);
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        license: Gold,
+      });
+      expect(mockResponse.forbidden).toBeCalled();
+    });
+    it('allows any license level to unisolate', async () => {
+      licenseEmitter.next(Gold);
+      await callRoute(UNISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        license: Gold,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+  });
+
+  describe('User Level', () => {
+    it('allows superuser to perform isolation', async () => {
+      const superU = { username: 'foo', roles: ['superuser'] };
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+    it('allows kibana_admin to perform isolation', async () => {
+      const superU = { username: 'foo', roles: ['kibana_admin'] };
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+
+    it('allows superuser to perform unisolation', async () => {
+      const superU = { username: 'foo', roles: ['superuser'] };
+      await callRoute(UNISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+    it('allows kibana_admin to perform unisolation', async () => {
+      const superU = { username: 'foo', roles: ['kibana_admin'] };
+      await callRoute(UNISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.ok).toBeCalled();
+    });
+
+    it('prohibits non-admin user from performing isolation', async () => {
+      const superU = { username: 'foo', roles: ['user'] };
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.forbidden).toBeCalled();
+    });
+    it('prohibits non-admin user from performing unisolation', async () => {
+      const superU = { username: 'foo', roles: ['user'] };
+      await callRoute(UNISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'] },
+        mockUser: superU,
+      });
+      expect(mockResponse.forbidden).toBeCalled();
+    });
+  });
+
+  describe('Cases', () => {
+    it.todo('logs a comment to the provided case');
```

Comment:
Don't forget about these :)

Reply:
implementation of this listed as todo in the PR comment!